code/game/objects/structures/window.dm

```diff
@@ -304,16 +304,19 @@
 		to_chat(G.assailant, SPAN_DANGER("You need a better grip to do that!"))
 		return TRUE
 	var/def_zone = ran_zone(BP_HEAD, 20)
+	var/mob/affecting_mob = G.get_affecting_mob()
```

Comment:
Maybe just call attackby if it's not a mob? Otherwise you get that using items in hand calls attackby, while using items pulled bashes with them.